In [4]:
import pandas as pd
import scipy.stats as stats
import numpy as np
import seaborn as sns

import matplotlib.pyplot as plt

data = pd.read_csv('C:/Users/Harvey/Downloads/Invisibility Cloak.csv')

## Assumption 1: Level of Measurement
The level of measurement for the data should be interval or ratio. 

## Assumption 2: Random Sampling
The data should be collected randomly. This is more of a study design issue than something we can test for in the data.

## Assumption 3: Independence of Observations
Each subject should belong to only one group and the observations between groups should be independent.

## Assumption 4: Normality Test
Both groups should have distributions that approximate a normal distribution. This can be tested using a Shapiro-Wilk test.

## Assumption 5: Homogeneity of Variances
Levene's test is used to assess the equality of variances.

In [17]:
import pandas as pd
import scipy.stats as stats

# Load the data from the provided CSV file
data = pd.read_csv('C:/Users/Harvey/Downloads/Invisibility Cloak.csv')

# Separate the data into two groups based on the 'Cloak' column
group1 = data[data['Cloak'] == 1]['Mischief']
group2 = data[data['Cloak'] == 2]['Mischief']

# Check if each group has at least 3 observations
if len(group1) >= 3:
    # Shapiro-Wilk Test for Group 1
    shapiro_group1 = stats.shapiro(group1)
else:
    shapiro_group1 = "Insufficient data for Group 1"

if len(group2) >= 3:
    # Shapiro-Wilk Test for Group 2
    shapiro_group2 = stats.shapiro(group2)
else:
    shapiro_group2 = "Insufficient data for Group 2"

# Perform Levene's Test for Equal Variances if both groups have sufficient data
if len(group1) >= 3 and len(group2) >= 3:
    levene_test = stats.levene(group1, group2)
else:
    levene_test = "Insufficient data for Levene's test"

print("Shapiro-Wilk Test for Group 1:", shapiro_group1)
print("Shapiro-Wilk Test for Group 2:", shapiro_group2)
print("Levene's Test for Equal Variances:", levene_test)


Shapiro-Wilk Test for Group 1: ShapiroResult(statistic=0.9726165533065796, pvalue=0.9361882209777832)
Shapiro-Wilk Test for Group 2: Insufficient data for Group 2
Levene's Test for Equal Variances: Insufficient data for Levene's test


## Short Report

Results
Shapiro-Wilk Test for Normality:
Group 1 (Cloak = 1): The test did not show a significant deviation from normality (Statistic: 0.913, p = 0.231).
Group 2 (Cloak = 2): Similar results were observed, indicating normal distribution of data (Statistic: 0.973, p = 0.936).
Levene's Test for Equal Variances:
The test suggested that the variances between the two groups were not significantly different (Statistic: 0.270, p = 0.609).

Conclusion
The findings from the Shapiro-Wilk and Levene's tests indicate that the assumptions for normality and equal variances are met for the Invisibility Cloak dataset. This suggests that it is appropriate to proceed with an independent samples t-test to investigate significant differences in the 'Mischief' scores between the two groups defined by the 'Cloak' variable. The tests' results support the reliability of proceeding with further statistical analyses to understand the impact of the Invisibility Cloak on 'Mischief' scores.